builders/reservations.go

```diff
@@ -0,0 +1,115 @@
+package builders
+
+import (
+	"encoding/json"
+	"fmt"
+
+	"github.com/pkg/errors"
+	"github.com/threefoldtech/tfexplorer"
+	"github.com/threefoldtech/tfexplorer/client"
+	"github.com/threefoldtech/tfexplorer/models/generated/workloads"
+	wrklds "github.com/threefoldtech/tfexplorer/pkg/workloads"
+	"github.com/threefoldtech/tfexplorer/schema"
+)
+
+// ReservationClient is a client to deploy and delete reservations
+type ReservationClient struct {
+	explorer *client.Client
+	userID   *tfexplorer.UserIdentity
+}
+
+// NewReservationClient creates a new reservation client
+func NewReservationClient(explorer *client.Client, userID *tfexplorer.UserIdentity) *ReservationClient {
+	return &ReservationClient{
+		explorer: explorer,
+		userID:   userID,
+	}
+}
+
+// Deploy deploys the reservation
+func (r *ReservationClient) Deploy(reservation workloads.Reservation, currencies []string) (wrklds.ReservationCreateResponse, error) {
+	res, err := r.DryRun(reservation, currencies)
+	if err != nil {
+		return wrklds.ReservationCreateResponse{}, nil
+	}
+
+	var reservationToCreate workloads.Reservation
+
+	err = json.Unmarshal(res, &reservationToCreate)
+	if err != nil {
+		return wrklds.ReservationCreateResponse{}, nil
+	}
+
+	fmt.Printf("%+v", reservationToCreate)
+
+	response, err := r.explorer.Workloads.Create(reservationToCreate)
+	if err != nil {
+		return wrklds.ReservationCreateResponse{}, errors.Wrap(err, "failed to send reservation")
+	}
+
+	return response, nil
+}
+
+// DryRun will return the reservation to deploy as JSOM
+func (r *ReservationClient) DryRun(reservation workloads.Reservation, currencies []string) ([]byte, error) {
+	userID := int64(r.userID.ThreebotID)
+	signer, err := client.NewSigner(r.userID.Key().PrivateKey.Seed())
+	if err != nil {
+		return nil, errors.Wrap(err, "could not load signer")
+	}
+
+	reservation.CustomerTid = userID
+	// we always allow user to delete his own reservations
+	reservation.DataReservation.SigningRequestDelete.QuorumMin = 1
+	reservation.DataReservation.SigningRequestDelete.Signers = []int64{userID}
```

Comment:
These 2 value should also be expose for configuration. This is where user can setup multisig for it's reservation.